src/solver/modeler/api/include/antares/solver/modeler/api/linearProblemFiller.h

```diff
@@ -30,9 +32,19 @@ namespace Antares::Solver::Modeler::Api
 class LinearProblemFiller
 {
 public:
-    virtual void addVariables(LinearProblem* problem, LinearProblemData* data) = 0;
-    virtual void addConstraints(LinearProblem* problem, LinearProblemData* data) = 0;
-    virtual void addObjectiveCoefficients(LinearProblem* problem, LinearProblemData* data) = 0;
+    explicit LinearProblemFiller(std::shared_ptr<ILinearProblem> pb, LinearProblemData& data):
+        linearProblem_(pb),
+        LPdata_(data)
+    {
+    }
+
+    virtual void addVariables() = 0;
+    virtual void addConstraints() = 0;
+    virtual void addObjective() = 0;
```

Comment:
I think we better keep the old signatures in order to allow the same fillers to be used to fill multiple matrices (LinearProblems), for example the same thermal filler can be used to fill multiple 1-week problems

Reply:
ok.
As sais in this PR's presentation and more accurately in [this (later) comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2391#discussion_r1758716311), there are drawbacks in doing what you ask : for dependency and simplicity reasons.

A solution to this could be to add a new abstract method to fillers interface : 
```C++
class LinearProblemFiller
{
public:
    explicit LinearProblemFiller(ILinearProblem& pb, LinearProblemData& data):
        linearProblem_(pb),
        LPdata_(data)
    {
    }
   virtual void resetLinearProblem(ILinearProblem& pb);
   virtual void addVariables() = 0;
   virtual void addConstraints() = 0;
   virtual void addObjective() = 0;

protected:
    ILinearProblem& linearProblem_;
    const LinearProblemData& LPdata_;
};
```

Reply:
I agree with @a-zakir 's comment that this should stay an interface and contain no attributes.
As for the drawbacks : 
- the three fillers' methods will only be called once in the builder, so it's not that big a drawback for them to have 2 arguments
- the builder actually is supposed to be the orchestrator of the LinearProblem and the Fillers (and that's its only purpose), so I'm willing to accept the dependencies

Reply:
ok. Done.